modules/porous_flow/src/ics/PorousFlowFluidPropertyIC.C

```diff
@@ -0,0 +1,69 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "PorousFlowFluidPropertyIC.h"
+#include "SinglePhaseFluidProperties.h"
+
+registerMooseObject("PorousFlowApp", PorousFlowFluidPropertyIC);
+
+template <>
+InputParameters
+validParams<PorousFlowFluidPropertyIC>()
+{
+  InputParameters params = validParams<InitialCondition>();
+  params.addRequiredCoupledVar("porepressure", "Fluid porepressure");
+  params.addRequiredCoupledVar("temperature", "Fluid temperature");
+  MooseEnum unit_choice("Kelvin=0 Celsius=1", "Kelvin");
+  params.addParam<MooseEnum>(
+      "temperature_unit", unit_choice, "The unit of the temperature variable");
+  params.addRequiredParam<UserObjectName>("fp", "The name of the user object for the fluid");
+  MooseEnum property_enum("enthalpy internal_energy density");
+  params.addRequiredParam<MooseEnum>(
+      "property", property_enum, "The fluid property that this auxillary kernel is to calculate");
```

Comment:
You mean "initial condition" not "auxiliary kernel", right?

Reply:
Bad copy and paste...